core-api/src/lib.rs

```diff
@@ -19,6 +20,119 @@ use temporal_sdk_core_protos::coresdk::{
+    ///
+    /// The returned activation is guaranteed to be for the same task queue / worker which was
+    /// provided as the `task_queue` argument.
+    ///
+    /// It is rarely a good idea to call poll concurrently. It handles polling the server
+    /// concurrently internally.
+    ///
+    /// TODO: Examples
+    async fn poll_activity_task(&self) -> Result<ActivityTask, PollActivityError>;
+
+    /// Tell the core that a workflow activation has completed. May be freely called concurrently.
+    async fn complete_workflow_activation(
+        &self,
+        completion: WorkflowActivationCompletion,
+    ) -> Result<(), CompleteWfError>;
+
+    /// Tell the core that an activity has finished executing. May be freely called concurrently.
+    async fn complete_activity_task(
+        &self,
+        completion: ActivityTaskCompletion,
+    ) -> Result<(), CompleteActivityError>;
+
+    /// Notify workflow that an activity is still alive. Long running activities that take longer
+    /// than `activity_heartbeat_timeout` to finish must call this function in order to report
+    /// progress, otherwise the activity will timeout and a new attempt will be scheduled.
+    ///
+    /// The first heartbeat request will be sent immediately, subsequent rapid calls to this
+    /// function will result in heartbeat requests being aggregated and the last one received during
+    /// the aggregation period will be sent to the server, where that period is defined as half the
+    /// heartbeat timeout.
+    ///
+    /// Unlike java/go SDKs we do not return cancellation status as part of heartbeat response and
+    /// instead send it as a separate activity task to the lang, decoupling heartbeat and
+    /// cancellation processing.
+    ///
+    /// For now activity still need to send heartbeats if they want to receive cancellation
+    /// requests. In the future we will change this and will dispatch cancellations more
+    /// proactively. Note that this function does not block on the server call and returns
+    /// immediately. Underlying validation errors are swallowed and logged, this has been agreed to
+    /// be optimal behavior for the user as we don't want to break activity execution due to badly
+    /// configured heartbeat options.
+    fn record_activity_heartbeat(&self, details: ActivityHeartbeat);
+
+    /// Request that a workflow be evicted by its run id. This will generate a workflow activation
+    /// with the eviction job inside it to be eventually returned by
+    /// [Core::poll_workflow_activation]. If the workflow had any existing outstanding activations,
+    /// such activations are invalidated and subsequent completions of them will do nothing and log
+    /// a warning.
+    fn request_workflow_eviction(&self, run_id: &str);
+
+    /// Returns this worker's instance of the [ServerGatewayApis] implementor it is using.
+    fn server_gateway(&self) -> Arc<dyn ServerGatewayApis + Send + Sync>;
```

Comment:
Should get call this `get_service_client`?

Reply:
This is returning the higher-level one, which lets you use the lower level one. I need this public for integ tests. Having both feels a bit overkill